src/lambda/config/templates.ts

```diff
@@ -395,3 +400,59 @@ export class TemplatesConfigPopulator {
         return this
     }
 }
+
+/**
+ * Returns a previously-configured event payload if the following are true (`undefined` if false):
+ * * old `.aws/templates.json` file exists and is valid JSON
+ * * handler exists in template registry
+ * * handler also exists in `.aws/templates.json`
+ * @param workspaceFolder Current workspace folder
+ * @param handler Function's handler
+ * @param samTemplate Originating SAM Template
+ */
+export async function getExistingConfiguration(
+    workspaceFolder: vscode.WorkspaceFolder,
+    handler: string,
+    samTemplate: vscode.Uri,
+    registry: CloudFormationTemplateRegistry = CloudFormationTemplateRegistry.getRegistry()
+): Promise<AdditionalDebuggerFields | undefined> {
+    const configPath: string = getTemplatesConfigPath(workspaceFolder.uri.fsPath)
+
+    if (await fsUtils.fileExists(configPath)) {
+        let foundHandler: boolean = false
+        for (const templateDatum of registry.registeredTemplates) {
```

Comment:
Is there a reason to not just look for the old file and convert it regardless of whether a handler is found/valid?

The intent is to move the user onto the new system. So I don't think it matters if their old stuff was "valid" as long as it matches the schema we expect.

Additionally, I would add an informational message to inform the user this occured.

Reply:
Are you proposing that we automatically move every single existing run config over automatically, whether or not they chose to move it? I'd rather not clutter a user's launch configs, especially if the config wouldn't work. If the majority would rather do that, then I can look to do that instead.

If you're proposing that we shouldn't ask users if they want to migrate, that's a decision for when we actually integrate this function.

Reply:
I'd be fine saying yes to both because there's parity and it's an upgraded experience for the user (more idiomatic).

If there's concern over choice, we could also give the option to opt out and also drop support for the files. Then possibly give the user an option to migrate the file over via another command.

Not really a fan of supporting both simultaneously if we can avoid it while offering an upgraded experience. It lets us move forward nicely without carrying all the baggage.

Reply:
I wouldn't consider this simultaneous support. I'm assuming that we'll remove any codelenses that let people alter the config or consume the config directly as part of a run/debug, and since this will piggyback on your change in #972, users won't even be able to create another run config with the template.yaml codelenses.

The only definite win I can think of is that we'd be able to blow away the `./.aws/templates.json` file since that'd be orphaned otherwise.

Reply:
Chatted offline with @awschristou and he's in agreement that we should only change on prompt and we should leave it up to the user to remove the old file when they're done with it.

Reply:
Ok I wasn't sure if you were going to support both. 

Not clear on the flow here. Would the user click the code lens and you would attempt to migrate one over if it's in the old file?

Why would you need to check that the handler is in the registry in that case?

Reply:
> Why would you need to check that the handler is in the registry in that case?

I've been doing some thinking about this and I think I had a better justification before I started thinking about it.

Ultimately, we want to ensure that the handler is in the template because these were specifically tied to templates before: codelenses were dependent on the handler being in the template.

At this point, this is more of a safety edge case if we have an access point to this functionality in the code itself (rather than in the template, where it's plainly obvious that we have a handler present): for instance, if a codelens is drawn in-code, the template is altered in a way that removes this, and the codelens doesn't update, in theory, it shouldn't do anything.

It's a fairly weak justification and it's one that we can probably work out on any code-side codelenses. I'm open to removing it.

Reply:
Yeah I understand wanting to verify its presence in the registry in general, but I don't think that check belongs with `getExistingConfiguration`. Seemingly that check would happen before this function is called. Presumably it would occur, as you've mentioned, when the command is executed as a sanity check.